plugin/src/main/groovy/com/novoda/staticanalysis/internal/VariantAware.groovy

```diff
@@ -0,0 +1,43 @@
+package com.novoda.staticanalysis.internal
+
+import org.gradle.api.DomainObjectSet
+import org.gradle.api.NamedDomainObjectSet
+
+trait VariantAware {
+
+    private boolean hasFilter = false
```

Comment:
why don't we keep out of the `trait` API and we just keep the `getFiltered*()`? this way we can choose in the configurator how to deal with that, for instance only Lint needs to know whether the filter is set or not, right?

Reply:
`trait` allows me to have implementation and fields here. I don't want to lose that. 

The other option is to make it nullable. Let me try and see how that goes actually. `hasFilter` is not necessary.

Reply:
Oh, addressing here, I pushed to `clean-up` branch. That looks much nicer.